pkg/types/baremetal/validation/platform.go

```diff
@@ -61,6 +61,66 @@ func validateOSImageURI(uri string) error {
 	return nil
 }
 
+// validateHosts checks that hosts have all required fields set with appropriate values
+func validateHosts(hosts []*baremetal.Host, fldPath *field.Path) field.ErrorList {
+	hostErrs := field.ErrorList{}
+
+	uniqueBMC := make(map[string]*baremetal.Host)
+	uniqueMAC := make(map[string]*baremetal.Host)
+	uniqueName := make(map[string]*baremetal.Host)
+
+	for _, host := range hosts {
+		// Ensure name is set
+		if host.Name == "" {
+			hostErrs = append(hostErrs, field.Invalid(fldPath.Child("hosts"), host.Name, "name is required"))
+		}
+
+		// Ensure name is unique
+		if other, ok := uniqueName[host.Name]; ok {
+			hostErrs = append(hostErrs, field.Invalid(fldPath.Child("hosts"), host.Name, fmt.Sprintf("duplicate name %s", other.Name)))
```

Comment:
i think you should
- have a outside the loop assignment for `fldPath = fldPath.Child("hosts")`
- and this one should be fldPath.Index($idx).Name

Reply:
also why not just use https://godoc.org/k8s.io/apimachinery/pkg/util/validation/field#Duplicate ??